src/github.com/getlantern/enproxy/conn_impl.go

```diff
@@ -160,9 +161,18 @@ func (c *conn) doRequest(proxyConn *connInfo, host string, op string, request *r
 	// Check response status
 	responseOK := resp.StatusCode >= 200 && resp.StatusCode < 300
 	if !responseOK {
-		err = fmt.Errorf("Bad response status for read: %s", resp.Status)
+		// This means we're getting something other than an OK response from the fronting provider
+		// itself, which is odd. Try to log the entire response for easier debugging.
+		full, er := httputil.DumpResponse(resp, true)
+		if er == nil {
```

Comment:
Even though it seems redundant or highly unlikely to fail, I think we should log also this error (currently we are only cheking if nil in order to use the dumped data or not). In any case, this will pass `errcheck`.